### PR TITLE
Enhancement: Add the ability to reset a task based concurrency limit

### LINKED ISSUE #1729: Add an option to reset task run concurrency limits from the UI

### PR DESCRIPTION
**Pull request by zhen0**
Closes #1729 

Brings the UI concurrency page up to parity with the cli by enabling reset of concurrency limits. @taljaards - this is one you requested! 

https://github.com/PrefectHQ/prefect-ui-library/assets/40272060/894fb13e-eb87-49e1-b11a-e278d89c05e7

### CODE REVIEW

File: src/components/ConcurrencyLimitMenu.vue [hunk 1]
```diff
@@ -2,13 +2,15 @@
   <p-icon-button-menu v-bind="$attrs">
     <copy-overflow-menu-item label="Copy ID" :item="concurrencyLimit.id" />
     <p-overflow-menu-item v-if="can.delete.concurrency_limit" label="Delete" @click="open" />
+    <p-overflow-menu-item v-if="can.delete.concurrency_limit" label="Reset" @click="openReset" />
```
Comment (pleek91) [inline src/components/ConcurrencyLimitMenu.vue:5]: Curious why the `delete` and not `update` permission check here. Assuming that's just following what the api allows?

File: src/components/ConcurrencyLimitMenu.vue [hunk 2]
```diff
@@ -37,6 +39,8 @@
 
   const { showModal, open, close } = useShowModal()
 
+  const { showModal: showResetModal, open: openReset, close: closeReset } = useShowModal()
```
Comment (pleek91) [inline src/components/ConcurrencyLimitMenu.vue:42]:
```suggestion
  const { showModal: showResetModal, open: openReset } = useShowModal()
```

**Review by pleek91 — APPROVED** [review]
One question and a linting warning but approving cause everything else looks good and this could go out as is.

File: src/components/ConcurrencyLimitMenu.vue (hunk 1 above)
Comment (zhen0) [reply src/components/ConcurrencyLimitMenu.vue]: Ah.  Good call!

**Pull request merged**